lms/djangoapps/courseware/views/views.py

```diff
@@ -937,14 +938,22 @@ def _get_cert_data(student, course, course_key, is_active, enrollment_mode):
         course = get_course_by_id(course_key)
         may_view_certificate = course.self_paced or course.may_certify()
 
-    show_message = all([
+    switches = certificates_waffle.waffle()
```

Comment:
Checking for the waffle switches. Only check if students can automatically generate their own certificates if the automatic cert generation is not turned on.